glusterfsd: fix all -Wmissing-field-initializers warnings

dmantipov opened the pull request:
Fix bunch of 77 -Wmissing-field-initializers warnings around
'struct argp_option' parameters of glusterfsd option tables.

Signed-off-by: Dmitry Antipov <dantipov@cloudlinux.com>
Updates: #1000

Commit:
glusterfsd: fix all -Wmissing-field-initializers warnings

Fix bunch of 77 -Wmissing-field-initializers warnings around
'struct argp_option' parameters of glusterfsd option tables.

Signed-off-by: Dmitry Antipov <dantipov@cloudlinux.com>
Updates: #1000

amarts commented:
/run regression

gluster-ant commented:
1 test(s) failed 
./tests/bugs/replicate/bug-986905.t

0 test(s) generated core 


4 test(s) needed retry 
./tests/000-flaky/glusterd-restart-shd-mux.t
./tests/bugs/bug-1371806_acl.t
./tests/bugs/replicate/bug-1297695.t
./tests/bugs/replicate/bug-986905.t
https://build.gluster.org/job/gh_centos7-regression/1353/

amarts commented:
/run regression

gluster-ant commented:
0 test(s) failed 


1 test(s) generated core 
./tests/basic/afr/afr-anon-inode.t

1 test(s) needed retry 
./tests/000-flaky/glusterd-restart-shd-mux.t
https://build.gluster.org/job/gh_centos7-regression/1359/

Commit:
glusterfsd: fix all -Wmissing-field-initializers warnings

Fix bunch of 77 -Wmissing-field-initializers warnings around
'struct argp_option' parameters of glusterfsd option tables.

Signed-off-by: Dmitry Antipov <dantipov@cloudlinux.com>
Updates: #1000

gluster-ant commented:
CLANG-FORMAT FAILURE:
Before merging the patch, this diff needs to be considered for passing clang-format

```diff --git a/glusterfsd/src/glusterfsd.c b/glusterfsd/src/glusterfsd.c
index 622676828..0571797fb 100644
--- a/glusterfsd/src/glusterfsd.c
+++ b/glusterfsd/src/glusterfsd.c
@@ -313,8 +313,7 @@ static struct argp_option gf_options[] = {
     {0, 0, 0, 0, "Miscellaneous Options:", 0},
     {
         0,
-    }
-};
+    }};
 
 static struct argp argp = {gf_options, parse_opts, argp_doc, gf_doc,
                            NULL,       NULL,       NULL};
@@ -1427,7 +1426,8 @@ parse_opts(int key, char *arg, struct argp_state *state)
         case ARGP_IO_ENGINE_KEY:
             cmd_args->io_engine = gf_strdup(arg);
             if (cmd_args->io_engine == NULL) {
-                argp_failure(state, -1, 0, "Failed to allocate memory for "
+                argp_failure(state, -1, 0,
+                             "Failed to allocate memory for "
                              "io-engine");
             }
             break;
@@ -2692,10 +2692,8 @@ out:
 int
 main(int argc, char *argv[])
 {
-    gf_io_handlers_t main_handlers = {
-        .setup = main_start,
-        .cleanup = main_terminate
-    };
+    gf_io_handlers_t main_handlers = {.setup = main_start,
+                                      .cleanup = main_terminate};
     glusterfs_ctx_t *ctx = NULL;
     int ret = -1;
     char cmdlinestr[PATH_MAX] = {
```

amarts commented:
/run regression
